kratos/spatial_containers/spatial_search_result.h

```diff
@@ -109,6 +116,39 @@ class SpatialSearchResult
     ///@name Access
     ///@{
 
+    /**
+    * @brief Get the ID of the object.
+    * @details If the object is derived from IndexedObject, retrieves its ID. Otherwise, returns 0.
+    * @return IndexType - The ID of the object or 0 if not derived from IndexedObject.
+    */
+    IndexType Id() const
+    {
+        // Get Id if the object is derived from IndexedObject
+        if constexpr (std::is_base_of_v<IndexedObject, TObjectType>) {
+            return mpObject->Id();
+        } else { // Otherwise 0
+            KRATOS_ERROR << "Object does not provide an ID" << std::endl;
+            return 0;
+        }
+    }
+
+    /**
+    * @brief Get the ID of the object.
+    * @details This function essentially performs the same operation as the Id() function. If the object is derived from IndexedObject, retrieves its ID. Otherwise, returns 0.
+    * @note Consider refactoring to avoid redundancy with Id() function.
+    * @return IndexType - The ID of the object or 0 if not derived from IndexedObject.
+    */
+    IndexType GetId() const
+    {
+        // Get Id if the object is derived from IndexedObject
+        if constexpr (std::is_base_of_v<IndexedObject, TObjectType>) {
```

Comment:
sorry but I would say that this is simply not correct ... the spatial search result is a global_pointer, so you cannot call directly a method on it as you don't know if it is local

Reply:
aside, if you need to call a function that requires an id onto an object without Id, i would say it is better not to overload the operator and to call it directly Id onto the result. If the function Id is not there it will give you a compile time error...